compiler/rustc_infer/src/infer/error_reporting/nice_region_error/mismatched_static_lifetime.rs

```diff
@@ -0,0 +1,84 @@
+
+        let (origin, sub, sup) = match error.clone() {
+            RegionResolutionError::ConcreteFailure(origin, sub, sup) => (origin, sub, sup),
+            _ => return None,
+        };
+        if *sub != ty::RegionKind::ReStatic {
+            return None;
+        }
+        let cause = match origin {
+            SubregionOrigin::Subtype(box TypeTrace { ref cause, .. }) => cause,
+            _ => return None,
+        };
+        let (parent, impl_def_id) = match &cause.code {
+            ObligationCauseCode::MatchImpl(parent, impl_def_id) => (parent, impl_def_id),
+            _ => return None,
+        };
+        let binding_span = match **parent {
+            ObligationCauseCode::BindingObligation(_def_id, binding_span) => binding_span,
+            _ => return None,
+        };
+        let impl_span = self.tcx().def_span(*impl_def_id);
+        let mut err = self.tcx().sess.struct_span_err(cause.span, "incompatible lifetime on type");
+        err.span_note(binding_span, "because of this predicate");
+        err.span_note(impl_span, "a compatible `impl` exists, but...");
+        note_and_explain_region(self.tcx(), &mut err, "", sup, "...");
+        note_and_explain_region(self.tcx(), &mut err, "...does not necessarily outlive ", sub, "");
+        if let Some(impl_local_def_id) = impl_def_id.as_local() {
+            let ty = self.tcx().type_of(*impl_def_id);
+            let mut v = super::static_impl_trait::TraitObjectVisitor(vec![]);
+            v.visit_ty(ty);
+            let matching_def_ids = v.0;
+
+            let impl_node =
+                self.tcx().hir().get(self.tcx().hir().local_def_id_to_hir_id(impl_local_def_id));
+            let impl_self_ty = match impl_node {
+                hir::Node::Item(hir::Item {
+                    kind: hir::ItemKind::Impl(hir::Impl { self_ty, .. }),
+                    ..
+                }) => self_ty,
+                _ => bug!("Node not an impl."),
+            };
+
+            for matching_def_id in matching_def_ids {
+                let mut hir_v =
+                    super::static_impl_trait::HirTraitObjectVisitor(vec![], matching_def_id);
+                hir_v.visit_ty(&impl_self_ty);
+                for span in &hir_v.0 {
+                    let mut multi_span: MultiSpan = vec![*span].into();
+                    multi_span.push_span_label(
+                        *span,
+                        "this has an implicit `'static` lifetime requirement".to_string(),
+                    );
+                    err.span_note(multi_span, "the used `impl` has a `'static` requirement");
+                    err.span_suggestion_verbose(
+                        span.shrink_to_hi(),
+                        "consider relaxing the implicit `'static` requirement",
+                        " + '_".to_string(),
+                        Applicability::MaybeIncorrect,
+                    );
+                }
```

Comment:
If there are multiple things adding `'static` lifetime requirements, this emits a single `note` instead of multiple.
```suggestion
                let mut multi_span: MultiSpan = vec![hir_v.0.clone()].into();
                for span in &hir_v.0 {
                    multi_span.push_span_label(
                        *span,
                        "this has an implicit `'static` lifetime requirement".to_string(),
                    );
                    err.span_suggestion_verbose(
                        span.shrink_to_hi(),
                        "consider relaxing the implicit `'static` requirement",
                        " + '_".to_string(),
                        Applicability::MaybeIncorrect,
                    );
                }
                err.span_note(multi_span, "the used `impl` has a `'static` requirement");
```